### PR TITLE
OAK-9801 improve error messages

### PR DESCRIPTION
**Pull request by joerghoh**
provide better error messages in case a null value is provided as parameter

### CODE REVIEW

**Review by mreutegg — COMMENTED** [review]
Looks useful to me. Can you please insert a space between the comma and the second parameter?

**Review by reschke — CHANGES_REQUESTED** [review]
Rather than adding more use of Guava, shouldn't we use https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String- ?

File: oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionImpl.java
Comment (anchela) [inline oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionImpl.java:222]: could we also mark the param as @NotNull in the jackrabbit API and here?

File: oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionImpl.java
Comment (anchela) [inline oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionImpl.java:178]: could we also mark the param as NotNull in the jackrabbit-api and here?

File: oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionImpl.java
Comment (anchela) [inline oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionImpl.java:195]: could we mark the param as NotNull in the jackrabbit-api and here? that might actually spare us the null check altogether..... couldn't it?

**Review by anchela — APPROVED** [review]
hi @joerghoh the changes look sensible to me.
what i would do though (probably under a different JIRA ticket) is mark the parameters for the getXOrNull methods as @NotNull to make the intent clear. then we might even omit the null-check altogether....

**Comment by joerghoh** [comment]
@anchela I agree on adding the annotations, but in the case of a public API I would do the checks nevertheless ...

**Comment by joerghoh** [comment]
I cannot merge, can someone from the Oak team perform that for me? Thanks!

**Comment by anchela** [comment]
hi @joerghoh , did you address the concern raised by @reschke ? once that is sorted out i will merge right away.

**Comment by joerghoh** [comment]
> Rather than adding more use of Guava, shouldn't we use https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String- ?

Good idea, switched over.

**Review by reschke — APPROVED** [review]
It would probably be good to undo white-space only changes...

**Comment by anchela** [comment]
hi @reschke , i usually agree that whitespace-only changes should be tracked separately. but since this PR is about a minor cleanup in the error-msg, i don't find them as distracting or making the PR harder to read. unless you object, i would merge it anyway.

**Comment by anchela** [comment]
@joerghoh , i merged it. for a next PR i would like to encourage you to follow the advice from @reschke to keep whitespace-only changes separately. it makes the PR harder to review as it distracts from the real changes you want to make.

**Pull request merged**
